symforce/slam/imu_preintegration/manifold_symbolic.py

```diff
@@ -0,0 +1,346 @@
+# ----------------------------------------------------------------------------
+# SymForce - Copyright 2022, Skydio, Inc.
+# This source code is under the Apache 2.0 license found in the LICENSE file.
+# ----------------------------------------------------------------------------
+
+import symforce.symbolic as sf
+from symforce import typing as T
+from symforce.jacobian_helpers import tangent_jacobians
+
+
+def internal_imu_residual(
+    pose_i: sf.Pose3,
+    vel_i: sf.V3,
+    pose_j: sf.Pose3,
+    vel_j: sf.V3,
+    accel_bias_i: sf.V3,
+    gyro_bias_i: sf.V3,
+    # Preintegrated measurements: state
+    DR: sf.Rot3,
+    Dv: sf.V3,
+    Dp: sf.V3,
+    # Other pre-integrated quantities
+    sqrt_info: sf.M99,
+    DR_D_gyro_bias: sf.M33,
+    Dp_D_accel_bias: sf.M33,
+    Dp_D_gyro_bias: sf.M33,
+    Dv_D_accel_bias: sf.M33,
+    Dv_D_gyro_bias: sf.M33,
+    # other
+    accel_bias_hat: sf.V3,
+    gyro_bias_hat: sf.V3,
+    gravity: sf.V3,
+    dt: T.Scalar,
+    epsilon: T.Scalar,
+) -> sf.Matrix:
```

Comment:
Would be good to annotate the exact shape of the return value here, and on the other symbolic functions in this PR where it's useful (I think most of them)

Reply:
Okay. Have done that.